src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1730,6 +1732,98 @@ export const handleReorderFormField = [
   _handleReorderFormField,
 ] as RequestHandler[]
 
+/**
+ * NOTE: Exported for testing.
+ * Private handler for PUT /forms/:formId/logic/:logicId
+ * @precondition Must be preceded by request validation
+ * @security session
+ *
+ * @returns 200 with success message when successfully updated
+ * @returns 403 when user does not have permissions to update logic
+ * @returns 404 when form cannot be found
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const _handleUpdateLogic: RequestHandler<
+  { formId: string; logicId: string },
+  unknown,
+  { updatedLogic: ILogicSchema }
```

Comment:
in addition, the request and response bodies should be typed as DTOs. you can look at other endpoints (e.g. update form field) as a reference

Reply:
edited